frontend/packages/ux-editor/src/components/config/expressions/DynamicContent.tsx

```diff
@@ -67,29 +82,58 @@ export const DynamicContent = ({ component, dynamic, onGetProperties, onAddDynam
     setExpressionElements(dynamic.expressionElements);
   };
 
-  const addExpressionElement = () => {
+  const addExpressionElement = (dynamicOperator: Operator) => {
     const newExpressionElement: ExpressionElement = { id: uuidv4() };
     const updatedExpressionElements = [...dynamic.expressionElements, newExpressionElement];
     dynamic.expressionElements.push(newExpressionElement);
+    setOperator(dynamicOperator);
+    dynamic.operator = dynamicOperator;
     setExpressionElements(updatedExpressionElements);
   };
 
+  const updateDynamicOperator = (dynamicOperator: Operator) => {
+    setOperator(dynamicOperator);
+  };
+
   const updateExpressionElement = () => {
     const updatedExpressionElements = [...dynamic.expressionElements];
     setExpressionElements(updatedExpressionElements);
   };
 
+  const handleUpdateComplexExpression = (newComplexExpression: any) => {
+    try {
+      const parsedComplexExpression = JSON.parse(newComplexExpression.replaceAll('\'', '\"'));
+      dynamic.complexExpression = JSON.parse(newComplexExpression.replaceAll('\'', '\"'));
+      setComplexExpression(parsedComplexExpression);
+    } catch (error) {
+      dynamic.complexExpression = newComplexExpression.length > 0 ? newComplexExpression : '[]';
+      setComplexExpression(newComplexExpression);
+    }
+  };
+
   const removeExpressionElement = (expressionElement: ExpressionElement) => {
-    if (dynamic.expressionElements.length < 2){
+    if (dynamic.expressionElements.length < 2) {
       const newExpressionElement: ExpressionElement = { id: uuidv4() };
       dynamic.expressionElements = [newExpressionElement];
       setExpressionElements([newExpressionElement]);
-    }
-    else {
+    } else {
       const updatedExpressionElements = dynamic.expressionElements.filter((expEl: ExpressionElement) => expEl !== expressionElement);
       const lastExpressionElement = updatedExpressionElements[updatedExpressionElements.length - 1];
-      dynamic.expressionElements = [...updatedExpressionElements.filter(expEl => expEl !== lastExpressionElement), { ...lastExpressionElement,  expressionOperatorForNextExpression: undefined }];
-      setExpressionElements([...updatedExpressionElements.filter(expEl => expEl !== lastExpressionElement), { ...lastExpressionElement,  expressionOperatorForNextExpression: undefined }]);
+      dynamic.expressionElements = [...updatedExpressionElements.filter(expEl => expEl !== lastExpressionElement), { ...lastExpressionElement }];
+      setExpressionElements([...updatedExpressionElements.filter(expEl => expEl !== lastExpressionElement), { ...lastExpressionElement }]);
```

Comment:
Hvorfor oppdaterer du både `dynamic`-objektet og staten på samme måte?

Reply:
Jeg mener det var for å synce objektet og det visuelle. 
Om jeg bare hadde oppdatert objektet så ville jo ikke det visuelle endret seg uten reload, og om jeg bare oppdaterer statens å vil jo ikke objektet få med seg endringen. Men veldig godt mulig jeg tar feil 😅
Eller tenker du på at jeg burde lagre "utregningen" av den nye staten i en const brukt den consten i state og på objekt?

Reply:
Det burde ikke være nødvendig å ha den samme informasjonen i to variabler. Hva brukes objektet til som state-variabelen ikke kan brukes til?